### PR TITLE
Added initial requestLEScan definitions for web-bluetooth

### PR DESCRIPTION
**Pull request by DaBs**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://webbluetoothcg.github.io/web-bluetooth/scanning.html
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/web-bluetooth/tsconfig.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
         "module": "commonjs",
         "lib": [
             "es6",
+            "ESNext",
```
Comment (thegecko) [inline types/web-bluetooth/tsconfig.json:6]: Why is this needed?

**Review by thegecko — CHANGES_REQUESTED** [review]
Happy with these, except it isn't clear why ESNext has been added. Can this be implemented without it?

File: types/web-bluetooth/tsconfig.json (hunk 1 above)
Comment (DaBs) [reply types/web-bluetooth/tsconfig.json:6]: This is down to .padStart being done in the test. I could probably just rewrite it, or specify it better as ES2017.String? It's not needed for the types themselves however
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

Comment (thegecko) [reply types/web-bluetooth/tsconfig.json:6]: You should be able to get a similar effect with something along the lines of:

```typescript
`0${b.toString(16)}`.slice(-2);
```

Comment (DaBs) [reply types/web-bluetooth/tsconfig.json]: Makes sense, I've changed it now. That being said, I'm a little curious, seeing as Web Bluetooth will anyway only be available in browsers that supports the ES2017.String spec, would there be an argument to considering those things in the future?

Thanks for the review though, much appreciated 😄

Comment (thegecko) [reply types/web-bluetooth/tsconfig.json]: Thanks for the changes, it's just always good to avoid extra prereqs/dependencies wherever possible IMO

**Review by thegecko — APPROVED** [review]
LGTM, thanks for the update

**Comment by DaBs** [comment]
Ready to merge
